core/dbt/parser/schema_yaml_readers.py

```diff
@@ -325,12 +349,15 @@ def _get_metric_type_params(self, type_params: UnparsedMetricTypeParams) -> Metr
             numerator=self._get_optional_metric_input(type_params.numerator),
             denominator=self._get_optional_metric_input(type_params.denominator),
             expr=str(type_params.expr) if type_params.expr is not None else None,
-            window=self._get_time_window(type_params.window),
+            window=self._get_optional_time_window(type_params.window),
             grain_to_date=grain_to_date,
             metrics=self._get_metric_inputs(type_params.metrics),
             conversion_type_params=self._get_optional_conversion_type_params(
                 type_params.conversion_type_params
-            )
+            ),
+            cumulative_type_params=self._get_optional_cumulative_type_params(
+                type_params.cumulative_type_params
+            ),
```

Comment:
I think either here or in `_get_optional_cumulative_type_params` we want to fall back on the deprecated values `type_params.window` and `type_params.grain_to_date` to populate the new values `type_params.cumulative_type_params.window` and `type_params.cululative_type_params.grain_to_date` respectively, if the new paths haven't been specified but the old paths have.